src/entities/data-address.ts

```diff
@@ -19,25 +19,6 @@ export interface HttpDataAddress extends BaseDataAddress {
   contentType?: string;
 }
 
-export interface S3StorageDataAddress extends BaseDataAddress {
```

Comment:
How come are we removing these specialised data addresses?

Reply:
The client should be general-purpose and not leverage on additional extensions, in fact the S3 and blobstorage extensions had been moved out of the EDC core repository (currently they are not maintained by the EDC committer group), a DataAddress is a free-form object, it's up to the implementors to built it in the correct way.
HttpDataAddress is in fact part of the core codebase so it's ok to have it pre-defined (I was uncertain if delete that as well tbh).
This is stuff that could be part of the "extensions" that we discussed about.

Reply:
This makes total sense! Thank you, Andrea. Looks good to me